src/Umbraco.Core/Collections/ObservableDictionary.cs

```diff
@@ -235,15 +236,76 @@ public bool TryGetValue(TKey key, out TValue val)
 
         #endregion
 
+        /// <summary>
+        /// The exception that is thrown when a duplicate key inserted.
+        /// </summary>
+        /// <seealso cref="System.Collections.ObjectModel.ObservableCollection{TValue}" />
+        /// <seealso cref="System.Collections.Generic.IReadOnlyDictionary{TKey, TValue}" />
+        /// <seealso cref="System.Collections.Generic.IDictionary{TKey, TValue}" />
+        [Serializable]
         internal class DuplicateKeyException : Exception
         {
+            /// <summary>
+            /// Gets the key.
+            /// </summary>
+            /// <value>
+            /// The key.
+            /// </value>
+            public string Key { get; }
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="DuplicateKeyException" /> class.
+            /// </summary>
+            public DuplicateKeyException()
+            { }
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="DuplicateKeyException" /> class.
+            /// </summary>
+            /// <param name="key">The key.</param>
             public DuplicateKeyException(string key)
-                : base("Attempted to insert duplicate key \"" + key + "\" in collection.")
+                : this(key, null)
+            { }
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="DuplicateKeyException" /> class.
+            /// </summary>
+            /// <param name="key">The key.</param>
+            /// <param name="innerException">The exception that is the cause of the current exception, or a null reference (<see langword="Nothing" /> in Visual Basic) if no inner exception is specified.</param>
+            public DuplicateKeyException(string key, Exception innerException)
+                : base("Attempted to insert duplicate key \"" + key + "\" in collection.", innerException)
             {
-                Key = key;
+                this.Key = key;
```

Comment:
We don't really use `this` coding pattern in this codebase, it's fine but i will remove it.

Reply:
I'll update this in the next commits 👍